lib/absinthe/federation/schema/entities_field.ex

```diff
@@ -160,4 +180,60 @@ defmodule Absinthe.Federation.Schema.EntitiesField do
         }
       }
     }
+
+  defp reduce_resolution(%{state: :resolved} = res), do: res.value
+
+  defp reduce_resolution(%{middleware: []} = res), do: res
+
+  defp reduce_resolution(%{middleware: [middleware | remaining_middleware]} = res) do
+    call_middleware(middleware, %{res | middleware: remaining_middleware})
+  end
+
+  defp call_middleware({Absinthe.Middleware.Dataloader, {loader, _fun}}, res) do
+    {source, typename} = find_relevant_dataloader(loader)
+
+    key_field =
+      Absinthe.Schema.lookup_type(res.schema, typename)
+      |> Absinthe.Type.meta()
+      |> Map.get(:key_fields, "id")
+
+    args =
+      get_in(
+        res,
+        [
+          Access.key(:arguments),
+          Access.key(:representations)
+        ]
+      )
+
+    ids =
+      args
+      |> Enum.reject(fn arg ->
+        Map.get(arg, "__typename") != typename
+      end)
+      |> Enum.map(fn arg -> Map.get(arg, key_field) end)
+
+    loader = Dataloader.load_many(loader, source, %{__typename: typename}, ids)
```

Comment:
```suggestion
    loader
      |> Dataloader.load_many(source, %{__typename: typename}, ids)
      |> Dataloader.run()
      |> Dataloader.get_many(source, %{__typename: typename}, ids)
      |> Enum.map(fn {_, data} -> data end)
```
You will need to remove the lines below as well with this change